### PR TITLE
Support timestamps in `parseRoute`

### PR DESCRIPTION
**Pull request by yakov116**
See https://github.com/sindresorhus/refined-github/pull/3135#issuecomment-633622883
Test URL's
https://github.com/sindresorhus/refined-github/blob/master@%7B2019-04-20T03:27:17Z%7D/license
https://github.com/sindresorhus/refined-github/blob/master@%7B2019-12-11T10:12:11Z%7D/source/features/comments-time-machine-links.tsx
https://github.com/sindresorhus/eslint-plugin-unicorn/tree/HEAD@%7B2020-05-24T16:33:08Z%7D

@fregante I dont have much time. (I did this very quickly)

### CODE REVIEW

**Comment by yakov116** [comment]
@fregante I didn't parse (pun intended) what you are saying.

**Comment by fregante** [comment]
There are many more formats supported by @: https://linux.die.net/man/1/git-rev-parse

What’s the issue in this case specifically?

**Comment by yakov116** [comment]
The see on default branch and latest tag will both fail on a time link.

Maybe just let it error out?

**Comment by fregante** [comment]
Neither. The problem is that the URL doesn't match the meta tag (`getCurrentBranch`) so the solution is to just skip this check and assume that the pathname passed matches the page.

File: source/github-helpers/parse-route.ts [hunk 1]
```diff
@@ -8,10 +8,17 @@ interface Pathname {
...
+	return timeStampRegex.test(decodeURIComponent(part1));
+}
+
 export default function parseRoute(pathname: string): Pathname {
 	const [user, repository, route, ...next] = pathname.replace(/^\/|\/$/g, '').split('/');
-	const parts = next.join('/');
 	const branch = getCurrentBranch();
+	next[0] = isTimeStamp(next[0]) ? branch + '/' : next[0];
+	const parts = next.join('/');
 	if (parts !== branch && !parts.startsWith(branch + '/')) {
```
Comment (fregante) [inline source/github-helpers/parse-route.ts:22]: Maybe 
```suggestion
	if (!isTimeStamp(next[0]) && parts !== branch && !parts.startsWith(branch + '/')) {
```

**Comment by fregante** [comment]
More test URLs:

https://github.com/sindresorhus/refined-github/blob/master@%7B1500%7D/package.json
https://github.com/sindresorhus/refined-github/blob/master@%7Byesterday%7D/package.json

Really there are a lot more situations where this might not work:

https://github.com/yakov116/refined-github/blob/master%5E/package.json
https://github.com/yakov116/refined-github/blob/master~1500/package.json

File: source/github-helpers/parse-route.ts (hunk 1 above)
Comment (yakov116) [reply source/github-helpers/parse-route.ts:22]: That will only work if there is no file name.
https://github.com/sindresorhus/eslint-plugin-unicorn/tree/HEAD@%7B2020-05-24T16:33:08Z%7D

But the following will not
https://github.com/sindresorhus/refined-github/blob/master@%7B2019-04-20T03:27:17Z%7D/license
https://github.com/sindresorhus/refined-github/blob/master@%7B2019-12-11T10:12:11Z%7D/source/features/comments-time-machine-links.tsx

**Comment by yakov116** [comment]
100% correct

File: source/github-helpers/parse-route.ts (hunk 1 above)
Comment (fregante) [reply source/github-helpers/parse-route.ts:22]: Just this

```suggestion
	// Exclude extended revisions from being tested #3137 https://git-scm.com/docs/git-rev-parse#_specifying_revisions
	if (parts !== branch && !parts.startsWith(branch + '/') && !/\^|~|@{/.test(next[0])) {
```

**Comment by fregante** [comment]
Wait I'll rewrite this function to be less strict

**Comment by yakov116** [comment]
Thanks. ❤️

**Pull request closed without merging**